src/functionlenses.jl

```diff
@@ -53,6 +53,13 @@ set(x,     ::typeof(imag), y) = real(x) + im*y
 set(x,    ::typeof(angle), y) = abs(x) * cis(y)
 set(x,      ::typeof(abs), y) = y >= zero(y) ? y * sign(x) : throw(DomainError(y, "cannot set abs($x) to $y"))
 
+set(x, ::typeof(modf), (fpart, ipart)) = ipart + fpart
+set(x, ::typeof(mod2pi), y) = set(x, Base.Fix2(mod, 2π), y)
+set(x, f::Base.Fix2{typeof(rem)}, y) = f.x * div(x, f.x) + y
+set(x, f::Base.Fix2{typeof(mod)}, y) = f.x * fld(x, f.x) + y
+set(x, f::Base.Fix2{typeof(div)}, y) = y * f.x + rem(x, f.x)
+set(x, f::Base.Fix2{typeof(fld)}, y) = y * f.x + mod(x, f.x)
```

Comment:
Could also add `mod1` and `cld`? 
Also should we check things like `0<=y < period` for `set(x, @optic(mod(_,period)), y)`?